crates/core/tedge_mapper/src/c8y/dynamic_discovery.rs

```diff
@@ -19,81 +21,54 @@ pub struct DiscoverOp {
 #[derive(thiserror::Error, Debug)]
 #[allow(clippy::enum_variant_names)]
 pub enum DynamicDiscoverOpsError {
-    #[error("Failed to add watch to directory: {0}")]
-    FailedtoAddWatch(String),
-
-    #[error("A non-UTF8 name cannot be used as an operation name: {0:?}")]
-    NotAnOperationName(OsString),
+    #[error("A non-UTF8 path cannot be parsed as an operation: {0:?}")]
+    NotAnOperation(PathBuf),
```

Comment:
Why changing the error message?

I think that the former message makes more sense for the end user. Parsing failure is a concern for the developer not for the user.